### DISCUSSION TITLE
Line between label and symbol

### DISCUSSION THREAD

**Discussion by qpincon** [maplibre/maplibre-gl-js#2999] [Ideas]
For clarity, I believe it would be nice to have the option to display a line between the label and the corresponding icon. Especially with offset and no overlap, it can become hard to read, but this option would help to clarify such situations, allowing even for larger offsets to have the opportunity to display more labels.

For instance on the [variable label placement page](https://maplibre.org/maplibre-gl-js/docs/examples/variable-offset-label-placement/), on a certain zoom level, it is not directly trivial which label belongs to what:
![before](https://github.com/maplibre/maplibre-gl-js/assets/10002904/8bdbc642-d78c-4c18-98e0-99e1ad64af5f)

Some lines would help:
![withlines](https://github.com/maplibre/maplibre-gl-js/assets/10002904/4b750143-88d8-4109-9af9-1eccc9f8def1)

 Some additions would be needed in the layer definition:
- `text-line-width`, in `layout` number, 0 (default) to disable
- `text-line-color`, in `paint`, to define the color of the line

What do you think?
Thank you!

**Comment by HarelM** [comment]
I personally think that if you need this, then you did something wrong. I think adding these lines is not a great user experience and so I would think what could've been done better to avoid the need for these lines as they make the map reading even harder...

**Reply by cns-solutions-admin** [reply]
I think that in cases with lots of icons and not enough space for labels this could improve readability.
Take for example our air picture (all aircrafts):

Compare the rich client with label lines:
![grafik](https://github.com/maplibre/maplibre-gl-js/assets/59872815/8dfcda3a-d5cd-4900-a493-165e077b7234)

to maplibre without lines:
![grafik](https://github.com/maplibre/maplibre-gl-js/assets/59872815/5ac0389e-05ee-44c4-a2a6-86cc80965ce9)

The map might be more cluttered with lines, but it is clear to which aircraft each label belongs.
